src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprAllRecipes.java

```diff
@@ -0,0 +1,197 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import ch.njol.util.coll.iterator.CheckedIterator;
+import org.bukkit.Bukkit;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Name("All Recipes")
+@Description({
+	"Retrieve all recipes registered in the server. Includes the options to:",
+	"<ul>",
+	"<li>Get only recipes provided by Minecraft or custom recipes</li>",
+	"<li>Specific recipe types</li>",
+	"<li>For specific items</li>",
+	"</ul>",
+	"",
+	"NOTES:",
+	"- When resetting the server recipes, all custom recipes from any plugin will be removed, regardless of specifying additional data. "
+		+ "Only vanilla recipes will be present.",
+	"- When deleting the server recipes, you are allowed to delete recipes using the options listed above.",
+})
+@Examples({
+	"set {_list::*} to all of the server's recipe of type shaped recipe for netherite ingot",
+	"set {_list::*} to the server's mc recipes of type cooking recipe for raw beef",
+	"set {_list::*} to server's custom recipes of type blasting for raw iron named \"Impure Iron\"",
+	"",
+	"reset all of the server's recipes",
+	"",
+	"delete all recipes for netherite ingot",
+	"clear all of the minecraft recipes",
+	"delete all of the custom shaped recipes"
+})
+@Since("INSERT VERSION")
+public class ExprAllRecipes extends SimpleExpression<Recipe> {
+
+	static {
+		Skript.registerExpression(ExprAllRecipes.class, Recipe.class, ExpressionType.SIMPLE,
+			"[all [of]] [the] server['s] recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] (mc|minecraft|vanilla) recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] custom recipes [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] (mc|minecraft|vanilla) %recipetype% [for %-itemstacks/itemtypes%]",
+			"[all [of]] [the] server['s] custom %recipetype% [for %-itemstacks/itemtypes%]");
```

Comment:
pickle pointed this one out to me recently, prevents `all of server...`
```suggestion
			"[all [[of] the]|the] server['s] recipes [for %-itemstacks/itemtypes%]",
			"[all [[of] the]|the] server['s] (mc|minecraft|vanilla) recipes [for %-itemstacks/itemtypes%]",
			"[all [[of] the]|the] server['s] custom recipes [for %-itemstacks/itemtypes%]",
			"[all [[of] the]|the] server['s] %recipetype% [for %-itemstacks/itemtypes%]",
			"[all [[of] the]|the] server['s] (mc|minecraft|vanilla) %recipetype% [for %-itemstacks/itemtypes%]",
			"[all [[of] the]|the] server['s] custom %recipetype% [for %-itemstacks/itemtypes%]");
```